cmd/rofl/deploy.go

```diff
@@ -249,6 +251,29 @@ var (
 	}
 )
 
+// detectTerm returns the preferred (longest) period of the given offer.
+func detectTerm(offer *roflmarket.Offer) (term roflmarket.Term) {
+	if offer == nil || offer.Payment.Native == nil {
+		cobra.CheckErr(fmt.Errorf("unable do determine payment term for offer '%s'", offer.ID))
```

Comment:
```suggestion
		cobra.CheckErr(fmt.Errorf("unable to determine payment term for offer '%s'", offer.ID))
```

Reply:
This will trigger a nil pointer dereference in case `offer == nil` and you want to get `offer.ID`.